### PR TITLE
Add a new test script for RTC time sync-up

### PR DESCRIPTION
**Pull request by juhlee-microsoft**
Added a new test case, POWER-CLOCK-SYNC-HIBERNATE
Fixed typos.
Adjust the code for the time stamp comparison and time values

**TEST RESULTS**

[LISAv2 Test Results Summary]
Test Run On           : 06/20/2020 16:50:50
ARM Image Under Test  : Canonical : UbuntuServer : 16.04-LTS : latest
Initial Kernel Version: 4.15.0-1089-azure
Final Kernel Version  : 5.8.0-rc1-next-20200618
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:53

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 POWER                POWER-CLOCK-SYNC-HIBERNATE                                                        PASS                51.18 

[LISAv2 Test Results Summary]
Test Run On           : 06/20/2020 16:50:48
ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
Initial Kernel Version: 5.3.0-1028-azure
Final Kernel Version  : 5.8.0-rc1-next-20200618
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:56

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 POWER                POWER-CLOCK-SYNC-HIBERNATE                                                        PASS                54.01

### CODE REVIEW

**Comment by juhlee-microsoft** [comment]
@dcui Please check if the logic makes sense,

**Comment by juhlee-microsoft** [comment]
@dcui I fixed the test script with timeout 12 minutes. Please check the logic/flow in the script and let me know if you have any comments. @LiliDeng @lubaihua33 @abhimarathe please review the PR.

**Comment by dcui** [comment]
> @dcui I fixed the test script with timeout 12 minutes. Please check the logic/flow in the script and let me know if you have any comments. @LiliDeng @lubaihua33 @abhimarathe please review the PR.

Thanks, Ju! The change looks good to me.

File: Testscripts/Windows/HB-CLOCKSYNC.ps1 [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+
+		# Configuration for the hibernation
+		Run-LinuxCmd -ip $AllVMData.PublicIP -port $AllVMData.SSHPort -username $user -password $password -command "./SetupHbKernel.sh" -RunInBackground -runAsSudo -ignoreLinuxExitCode:$true | Out-Null
+		Write-LogInfo "Executed SetupHbKernel script inside VM"
+
+		# Wait for kernel compilation completion. 90 min timeout
+		$timeout = New-Timespan -Minutes $maxKernelCompileMin
+		$sw = [diagnostics.stopwatch]::StartNew()
+		while ($sw.elapsed -lt $timeout){
+			$vmCount = $AllVMData.Count
```
Comment (LiliDeng) [inline Testscripts/Windows/HB-CLOCKSYNC.ps1]: ditto in another PR

File: Testscripts/Windows/HB-CLOCKSYNC.ps1 [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+		$12minAfterTimeStamp = Get-Content $LogDir\12min_after_timestamp.log
+		Write-LogDbg $12minAfterTimeStamp
+
+		if ($beforeTimeStamp -ne $12minAfterTimeStamp) {
+			Write-LogInfo "Successfully verified the beforeTimeStamp was different from 12minAfterTimeStamp"
+		} else {
+			Write-LogErr "Did not find time synced. $beforeTimeStamp to $12minAfterTimeStamp"
+		}
+
+		$controllerTimeStamp = Get-Date -format "yyyy"
```
Comment (LiliDeng) [inline Testscripts/Windows/HB-CLOCKSYNC.ps1:285]: just compare the year?

File: Testscripts/Windows/HB-CLOCKSYNC.ps1 [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+		Write-LogDbg $controllerTimeStamp
+
+		if ($12minAfterTimeStamp -eq $controllerTimeStamp) {
+			Write-LogInfo "Successfully verified the system date changed back to correct date"
+		} else {
+			Write-LogErr "Expected VM time changed back to the correct one after sync-up, but found $12minAfterTimeStamp"
```
Comment (LiliDeng) [inline Testscripts/Windows/HB-CLOCKSYNC.ps1]: when hit the error, the test should fail?

File: Testscripts/Windows/HB-CLOCKSYNC.ps1 [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+
+		$beforeTimeStamp = Get-Content $LogDir\before_timestamp.log
+		Write-LogDbg $beforeTimeStamp
+		$12minAfterTimeStamp = Get-Content $LogDir\12min_after_timestamp.log
+		Write-LogDbg $12minAfterTimeStamp
+
+		if ($beforeTimeStamp -ne $12minAfterTimeStamp) {
+			Write-LogInfo "Successfully verified the beforeTimeStamp was different from 12minAfterTimeStamp"
+		} else {
+			Write-LogErr "Did not find time synced. $beforeTimeStamp to $12minAfterTimeStamp"
```
Comment (LiliDeng) [inline Testscripts/Windows/HB-CLOCKSYNC.ps1]: when hit the fail, not change the test result?

**Review by LiliDeng — CHANGES_REQUESTED** [review]

File: Testscripts/Windows/HB-CLOCKSYNC.ps1 (hunk 2 above)
Comment (juhlee-microsoft) [reply Testscripts/Windows/HB-CLOCKSYNC.ps1:285]: Yes. Setting date (Month and Day) not important and Day are changing inside back and forth. Like new setting goes like 12:00:00, then the actual set in the kernel comes down to 11:59:59, and then it affect to days value. It looks there is a few ms offset in the setting and actual hw clock.

File: Testscripts/Windows/HB-CLOCKSYNC.ps1 (hunk 1 above)
Comment (juhlee-microsoft) [reply Testscripts/Windows/HB-CLOCKSYNC.ps1]: Fixed in e3b25b3b590e7e0d5575dc6cb1e39e2289eaa1db

File: Testscripts/Windows/HB-CLOCKSYNC.ps1 (hunk 3 above)
Comment (juhlee-microsoft) [reply Testscripts/Windows/HB-CLOCKSYNC.ps1]: Fixed in 3d27b72ed5f302940df8a778f189649817382860

File: Testscripts/Windows/HB-CLOCKSYNC.ps1 (hunk 4 above)
Comment (juhlee-microsoft) [reply Testscripts/Windows/HB-CLOCKSYNC.ps1]: Fixed in 3d27b72ed5f302940df8a778f189649817382860. We were not sure the estimated interval until next time sync. Found it later.

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**